jenkins/PerfCI_Chaos/03_PerfCI_Choas_Windows_VMs_Deployment/Jenkinsfile

```diff
@@ -109,6 +109,33 @@ END
             }
         }
 
+        stage('Prepare Google credential files') {
+            steps {
+                    script {
+                        // Assign the file path to the environment variable
+                        env.GOOGLE_DRIVE_CREDENTIALS = "${env.WORKSPACE}/credentials.json"
+                        env.GOOGLE_DRIVE_TOKEN = "${env.WORKSPACE}/token.json"
+
+                        sh """
+                        echo "Checking if ${env.GOOGLE_DRIVE_CREDENTIALS} exists..."
+                        if [ -f "${env.GOOGLE_DRIVE_CREDENTIALS}" ]; then
+                            echo "File exists. Deleting it..."
+                            rm -f "${env.GOOGLE_DRIVE_CREDENTIALS}"
+                        fi
+                        echo "Checking if ${env.GOOGLE_DRIVE_TOKEN} exists..."
+                        if [ -f "${env.GOOGLE_DRIVE_TOKEN}" ]; then
+                            echo "File exists. Deleting it..."
+                            rm -f "${env.GOOGLE_DRIVE_TOKEN}"
+                        fi
+
+                        echo "Copying secret file to the temporary local file..."
+                        cp "$GOOGLE_DRIVE_CREDENTIALS_FILE" "${env.GOOGLE_DRIVE_CREDENTIALS}"
+                        cp "$GOOGLE_DRIVE_TOKEN_FILE" "${env.GOOGLE_DRIVE_TOKEN}"
```

Comment:
Should the permissions be restrictive (e. g. 0600)?  You might want to put

```
umask 77
```
in the script.